### PR TITLE
docs: expand description

### PR DESCRIPTION
**Pull request by lucab**
This improves documentation and re-format the codebase:
 * added a rustdoc crate summary and unify examples
 * expanded README description to match rustdoc
 * run rustfmt on the whole codebase

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,49 @@
...
+//! [reproducible]: http://0pointer.net/blog/projects/stateless.html
+//!
+//! # Example
+//!
+//! ```rust,no_run
+//! # use liboverdrop::FragmentScanner;
+//! // Scan for fragments under:
+//! //  * /usr/lib/my-crate/config.g/*.toml
+//! //  * /run/my-crate/config.g/*.toml
+//! //  * /etc/my-crate/config.g/*.toml
```
Comment (rfairley) [inline src/lib.rs]: Typo: `.g` -> `.d`

**Review by rfairley — APPROVED** [review]
LGTM! Just one minor nit.

**Comment by lucab** [comment]
@rfairley good catch! Typo fixed, mind a last pass+merge?

**Comment by rfairley** [comment]
LGTM, merging!

**Comment by lucab** [comment]
May I tag a 0.0.2 with the new docstrings?

**Comment by rfairley** [comment]
@lucab yes, good with me!

**Pull request merged**